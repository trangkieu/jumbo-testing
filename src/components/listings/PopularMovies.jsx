import Grid from "./Grid";
import React from "react";
import Search from "./Search";

export const PopularMovies = ({popularMovies, detailsHook, resultsHook}) => {
    const popularMoviesDisplay = popularMovies !== null ?
        <Grid items={popularMovies} maxDisplay={6}
              detailsHook={detailsHook}
        /> : <div/>;

    return (
        <div>
            <Search resultsHook={resultsHook}/>
            <div id="popular-movies" >Popular Movies</div>
            <div>
                {popularMoviesDisplay}
            </div>
        </div>
    )
}