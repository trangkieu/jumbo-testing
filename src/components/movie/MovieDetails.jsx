import React from "react";
import moment from "moment";

export const MovieDetails = ({details}) => {

    const releaseDate = new Date(moment(details.release_date, "YYYY-MM-DD"));
    const votePercentage = Number(details.vote_average).toFixed(2) * 10;
    const runTime = Number(details.runtime);
    const runTimeHour = Math.floor(runTime / 60);
    const runTimeMinutes = runTime % 60;
    return (


        <div className="details">
            <div className="details-header">
                <div className="poster-details">
                    <img id="posterImg" src={'http://image.tmdb.org/t/p/original' + details.poster_path}
                         alt="poster"/>
                </div>
                <div className="details-section">
                    <div id="movie-title" className="movie-title">{details.title}</div>
                    <div id="rating" className="large-font">{releaseDate.getFullYear()} &#183; {votePercentage}% User Score</div>
                    <div id="runtime" className="large-font">{runTimeHour}hr {runTimeMinutes}min</div>
                </div>

                <div className="backdrop-details">
                    <img id="backdropImg" src={'http://image.tmdb.org/t/p/original' + details.backdrop_path}
                         alt="backdrop"/>
                </div>
            </div>
            <div className="overview">
                <div className="overview-title">Overview</div>
                <div id="movieDescriptions" className="overview-desc"><p>{details.overview}</p></div>

            </div>
        </div>

    )
}