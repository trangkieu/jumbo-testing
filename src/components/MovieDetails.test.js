import renderer from "react-test-renderer";
import {MovieDetails} from "./listings/MovieDetails";
import mockMovieDetails from "../data/movieDetails";
import React from "react";

describe('Movie details', () => {

    const component = renderer.create(<MovieDetails details={mockMovieDetails}/>);
    test('snapshot renders', () => {
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    // verify image urls
    const images = component.root.findAllByType('img');
    let backdropImageCount = 0;
    let posterImageCount = 0;
    images.map(img => {
        it('backdrop image ', () => {
            if (img.props.id === "backdropImg") {
                expect(img.props.src).toEqual('http://image.tmdb.org/t/p/original' + '/93xA62uLd5CwMOAs37eQ7vPc1iV.jpg');
                backdropImageCount++;
            }

        })
        it('poster image ', () => {
            if (img.props.id === "posterImg") {
                expect(img.props.src).toEqual('http://image.tmdb.org/t/p/original' + '/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg');
                posterImageCount++;
            }
        })
    });
    it('backdrop image counts', () => {
        expect(backdropImageCount).toEqual(1);
    });
    it('poster image counts', () => {
        expect(posterImageCount).toEqual(1);
    });

    //verify other text data
    const divs = component.root.findAllByType('div');
    let yearReleaseCount = 0;
    let votePercentageCount = 0;
    let runtimeCount = 0;

    it('year release details', () =>{
        divs.map( d => {
            if (d.props.id === 'yearRelease') {
                expect(d.props.children).toEqual(2018);
                yearReleaseCount++;
            }
            if(d.props.id === 'votePercentage') {
                expect(d.props.children.toString().replace(/,/g,'')).toEqual("80% User Score");
                votePercentageCount++;
            }
            if(d.props.id === 'runtime') {
                expect(d.props.children.toString().replace(/,/g,'')).toEqual("2hr 15min");
                runtimeCount++;
            }
            if(d.props.id === 'movieDescriptions') {
                const descriptions = d.props.children;
                const titleChild = descriptions.props.children[0];
                const overviewChild = descriptions.props.children[1];
                it('movie title ', () => {
                    expect(titleChild.props.children.toString().replace(/,/g,'')).toEqual("Bohemian Rhapsody");

                });
                it('movie overiew', () => {
                    expect(overviewChild.props.children.toString().replace(/,/g,'')).toEqual("Singer Freddie Mercury guitarist Brian May drummer Roger Taylor and bass guitarist John Deacon take the music world by storm when they form the rock 'n' roll band Queen in 1970. Hit songs become instant classics. When Mercury's increasingly wild lifestyle starts to spiral out of control Queen soon faces its greatest challenge yet â€“ finding a way to keep the band together amid the success and excess.");

                })
            }

        })
    })

    it('year release count ', () => {
        expect(yearReleaseCount).toEqual(1);
    })
    it('vote percentage count ', () => {
        expect(votePercentageCount).toEqual(1);
    })
    it('runtime percentage count ', () => {
        expect(runtimeCount).toEqual(1);
    })
});