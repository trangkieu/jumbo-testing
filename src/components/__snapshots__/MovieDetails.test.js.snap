exports[`Movie details snapshot renders 1`] = `
<div
  className="details">
  <div
    className="details-header">
    <div
      className="poster-details">
      <img
        alt="poster"
        id="posterImg"
        src="http://image.tmdb.org/t/p/original/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg" />
    </div>
    <div
      className="details-section">
      <div
        className="movie-title"
        id="movie-title">
        Bohemian Rhapsody
      </div>
      <div
        className="large-font"
        id="rating">
        2018
         · 
        80
        % User Score
      </div>
      <div
        className="large-font"
        id="runtime">
        2
        hr 
        15
        min
      </div>
    </div>
    <div
      className="backdrop-details">
      <img
        alt="backdrop"
        id="backdropImg"
        src="http://image.tmdb.org/t/p/original/93xA62uLd5CwMOAs37eQ7vPc1iV.jpg" />
    </div>
  </div>
  <div
    className="overview">
    <div
      className="overview-title">
      Overview
    </div>
    <div
      className="overview-desc"
      id="movieDescriptions">
      <p>
        Singer Freddie Mercury, guitarist Brian May, drummer Roger Taylor and bass guitarist John Deacon take the music world by storm when they form the rock \'n\' roll band Queen in 1970. Hit songs become instant classics. When Mercury\'s increasingly wild lifestyle starts to spiral out of control, Queen soon faces its greatest challenge yet – finding a way to keep the band together amid the success and excess.
      </p>
    </div>
  </div>
</div>
`;
